### PR TITLE
Python: rework query parameters

### LINKED ISSUE #1504: Python: split query parameters into multiple structs
https://github.com/iotaledger/iota-sdk/pull/1428

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Python: rework query parameters

## Links to any relevant issues

Closes #1504 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Running example (fails because node is not healthy, but no serialization error or something before)

### CODE REVIEW

File: bindings/python/iota_sdk/client/_node_indexer_api.py [hunk 1]
```diff
@@ -31,109 +31,241 @@ class NodeIndexerAPI(metaclass=ABCMeta):
...
+        """
+        page_size: Optional[int] = None
+        cursor: Optional[str] = None
+        created_before: Optional[int] = None
+        created_after: Optional[int] = None
+
+    @json
+    @dataclass
+    class OutputQueryParameters(CommonQueryParameters):
+        """Common Query parameters
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/client/_node_indexer_api.py]: no

File: bindings/python/iota_sdk/client/_node_indexer_api.py [hunk 2]
```diff
@@ -31,109 +31,241 @@ class NodeIndexerAPI(metaclass=ABCMeta):
...
         **Attributes:**
-        address :
+
+        page_size:
+            The maximum amount of items returned in one call. If there are more items, a cursor to the next page is
+            returned too. The parameter is ignored when pageSize is defined via the cursor parameter.
+        cursor:
+            Starts the search from the cursor (createdSlotIndex+outputId.pageSize).
+            If an empty string is provided, only the first page is returned.
+        created_after:
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/client/_node_indexer_api.py]: put it after before

File: bindings/python/iota_sdk/client/_node_indexer_api.py [hunk 3]
```diff
@@ -31,109 +31,241 @@ class NodeIndexerAPI(metaclass=ABCMeta):
...
+            If an empty string is provided, only the first page is returned.
+        created_before:
+            Returns outputs that were created before a certain slot index.
+        created_after:
+            Returns outputs that were created after a certain slot index.
+        """
+        page_size: Optional[int] = None
+        cursor: Optional[str] = None
+        created_before: Optional[int] = None
+        created_after: Optional[int] = None
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/client/_node_indexer_api.py]: We have SlotIndex in Python

Comment (Thoralf-M) [reply bindings/python/iota_sdk/client/_node_indexer_api.py]: Thanks

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**